### PR TITLE
Fix issue - custom-kernel: Install fails if the VMs have a static name

### PR DESCRIPTION
**Pull request by LiliDeng**
Resolve #662

### CODE REVIEW

File: Libraries/CommonFunctions.psm1 [hunk 1]
```diff
@@ -492,7 +492,11 @@ Function Provision-VMsForLisa($allVMData, $installPackagesOnRoleNames)
 				$jobID = Run-LinuxCmd -ip $vmData.PublicIP -port $vmData.SSHPort -username "root" -password $password -command "/root/$scriptName" -RunInBackground
 				$packageInstallObj = New-Object PSObject
 				Add-member -InputObject $packageInstallObj -MemberType NoteProperty -Name ID -Value $jobID
-				Add-member -InputObject $packageInstallObj -MemberType NoteProperty -Name RoleName -Value $vmData.RoleName
+				if($vmData.RoleName.Contains($vmData.ResourceGroupName)) {
+					Add-member -InputObject $packageInstallObj -MemberType NoteProperty -Name RoleName -Value $vmData.RoleName
```
Comment (lizzha) [inline Libraries/CommonFunctions.psm1]: why not change the role name when VM is deployed?

Comment (LiliDeng) [reply Libraries/CommonFunctions.psm1]: this value just used within this method and used for rename the build-kernel log file

**Comment by LiliDeng** [comment]
Testing is in progress.

image

**Comment by LiliDeng** [comment]
[LISAv2 Test Results Summary]
Test Run On           : 01/29/2019 12:11:02
ARM Image Under Test  : Canonical : UbuntuServer : 16.04-LTS : latest
Test Category         : Functional
Test Area             : SRIOV
Total Test Cases      : 11 (10 Pass, 0 Fail, 1 Abort)
Total Time (dd:hh:mm) : 0:3:50

   ID TestCaseName                                                                TestResult TestDuration(in minutes) 
---------------------------------------------------------------------------------------------------------------------
    1 SRIOV-VERIFY-VF-MAX-CONNECTION-MAX-VCPU                                        Aborted                    0 
    2 SRIOV-VERIFY-VF-BASIC-CONNECTION                                                  PASS                 6.77 
    3 SRIOV-IPERF-STRESS                                                                PASS                38.77 
    4 SRIOV-DISABLEVF-ON-GUEST                                                          PASS                 6.89 
    5 SRIOV-RELOAD-MODULE                                                               PASS                 6.92 
    6 VERIFY-ENABLE-DISABLE-SRIOV-ON-EXISTING-VM                                        PASS                 29.3 
	EnableSRIOV : Test Iteration - 1 : PASS 
	DisableSRIOV : Test Iteration - 1 : PASS 
	EnableSRIOV : Test Iteration - 2 : PASS 
	DisableSRIOV : Test Iteration - 2 : PASS 
	EnableSRIOV : Test Iteration - 3 : PASS 
	DisableSRIOV : Test Iteration - 3 : PASS 
	EnableSRIOV : Test Iteration - 4 : PASS 
	DisableSRIOV : Test Iteration - 4 : PASS 
	EnableSRIOV : Test Iteration - 5 : PASS 
	DisableSRIOV : Test Iteration - 5 : PASS 
    7 SRIOV-VERIFY-VF-BASIC-CONNECTION-MAX-VCPU                                         PASS                 6.63 
    8 VERIFY-IFUP-NICS-SRIOV                                                            PASS                 2.77 
    9 SRIOV-VERIFY-VF-MAX-CONNECTION                                                    PASS                  9.8 
   10 VERIFY-SRIOV-LSPCI                                                                PASS                 3.16 
   11 SRIOV-VERIFY-VF-MULTIPLE-CONNECTION                                               PASS                 7.07 

[LISAv2-Test-GJ75.log](https://github.com/LIS/LISAv2/files/2810224/LISAv2-Test-GJ75.log)

File: Libraries/CommonFunctions.psm1 [hunk 2]
```diff
@@ -1327,7 +1343,7 @@ Function Test-SRIOVInLinuxGuest {
 	while ($retValue -eq $false -and $Attempts -le $MaximumAttempts) {
 		Write-LogInfo "[Attempt $Attempts/$MaximumAttempts] Detecting Mellanox NICs..."
 		$DetectedSRIOVNics = Run-LinuxCmd -username $username -password $password -ip $IpAddress -port $SSHPort -command $VerificationCommand -runAsSudo
-		$DetectedSRIOVNics = [int]$DetectedSRIOVNics
+		$DetectedSRIOVNics = [int]$DetectedSRIOVNics[-1].ToString()
```
Comment (LiliDeng) [inline Libraries/CommonFunctions.psm1]: This is to resolve the issue found in pipeline, when return values like below format.

$DetectedSRIOVNics
sudo: unable to resolve host LISAv2-OneVM-lilitest-PE40-20190129011742-role-0
1

**Comment by lizzha** [comment]
Why is the case aborted?

**Comment by LiliDeng** [comment]
Provision issue, see serial log

[dependency-vm.9cf03f40-db50-4ec6-93d7-14e34fe76470.serialconsole.log](https://github.com/LIS/LISAv2/files/2810690/dependency-vm.9cf03f40-db50-4ec6-93d7-14e34fe76470.serialconsole.log)

**Comment by LiliDeng** [comment]
@adriansuhov Please review again.

**Pull request merged**